react/IOMessage.tsx

```diff
@@ -14,11 +26,39 @@ const IOMessage: IOMessageType = ({ children, id, ...props }) => {
     )
   }
 
+  /**
+   * This enables a user to pass a translatable message which is not
+   * in the current IntlContext to the function and actually have it
+   * formatted by react-intl.
+   */
+  const cache = createIntlCache()
+  const newIntl = createIntl(
+    {
+      locale: intl.locale,
+      messages: {
+        [id]: id,
+      },
+    },
+    cache
+  )
```

Comment:
I think you can extract this to a function in a utils file since this is used in two different places